tests/test_jsvalidation.py

```diff
@@ -0,0 +1,597 @@
+            "billing_address": {"$ref": "#/definitions/differentaddress"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        },
+                        "required": ["type"]
+                    }
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(jsonschema.exceptions.RefResolutionError):
+        obj.validate(v1, 'first')
+
+
+def test_ref_conflict_between_schemas():
+    """Test that having a conflict between schemas with references passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"},
+                    "city": {"type": "string"},
+                    "state": {"type": "string"}
+                },
+                "required": ["street_address", "city", "state"]
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        },
+                        "required": ["type"]
+                    }
+                ]
+            }
+        }
+    }
+    v2 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "differenttype"},
+                    "city": {"type": "string"},
+                    "state": {"type": "string"}
+                },
+                "required": ["street_address", "city", "state"]
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        },
+                        "required": ["type"]
+                    }
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v2, 'second')
+
+
+def test_none_types_pass():
+    """Test that having a field with no type passes."""
```

Comment:
Sorry I meant add:
```FIXME: Later we will want to add an optional failure for unknown types.```

It is always better to have a message with the FIXME so that the next person to see it knows what needs to be fixed. We will also create an issue for this once the PR is merged.